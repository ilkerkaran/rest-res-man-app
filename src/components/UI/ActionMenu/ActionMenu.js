import React from 'react';
import PropTypes from 'prop-types';
import {
  Menu, IconButton, MenuItem, Box
} from '@material-ui/core';
import { MoreVert as MoreVertIcon } from '@material-ui/icons';
import { makeStyles } from '@material-ui/core/styles';
import { selectOptionTypes } from '../../../propTypes/types';

const useStyles = makeStyles({
  actionMenu: {
    display: 'flex',
    justifyContent: 'flex-end',
    width: '95%'
  }
});
const ActionMenu = ({ icon, options }) => {
  const [anchorEl, setAnchorEl] = React.useState(null);
  const classes = useStyles();

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = (value) => {
    setAnchorEl(null);
    const selectedOption = options.filter((o) => o.event).find(
      (o) => o.value === value
    );
    if (selectedOption && selectedOption.event) { selectedOption.event(value); }
  };

  const optionsContent = options.filter((o) => o.event).map((option) => (
    <MenuItem key={option.value} onClick={() => handleClose(option.value)}>
      {option.text}
    </MenuItem>
  ));
  return optionsContent.length > 0 ? (
    <>
      <Box className={classes.actionMenu}>
        <IconButton onClick={handleClick}>
          {icon || <MoreVertIcon />}
        </IconButton>
        <Menu
          anchorEl={anchorEl}
          keepMounted
          open={!!anchorEl}
          onClose={handleClose}
        >
          {optionsContent}
        </Menu>
      </Box>
    </>
  ) : null;
};

export default ActionMenu;
ActionMenu.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape(selectOptionTypes)).isRequired,
  icon: PropTypes.object
};

ActionMenu.defaultProps = {
  icon: null

};
