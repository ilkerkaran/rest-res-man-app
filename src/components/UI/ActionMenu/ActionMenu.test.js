/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { render } from 'enzyme'
import ActionMenu from './ActionMenu'

describe('<ActionMenu /> ', () => {
  const fieldProps = {
    options: []
  }
  let wrapper
  const Composition = () => <ActionMenu {...fieldProps} />

  beforeEach(() => {
    wrapper = render(<Composition />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
