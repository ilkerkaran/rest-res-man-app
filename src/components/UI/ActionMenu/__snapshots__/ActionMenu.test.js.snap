// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionMenu />  should match snapshot 1`] = `
Object {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
